### PR TITLE
Redux persist

### PR DESCRIPTION
**Pull request by amalexandrescu**
Adds redux persist for remembering the theme on refresh of the page

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -5,10 +5,7 @@
     <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
     <meta name="viewport" content="width=device-width, initial-scale=1" />
     <meta name="theme-color" content="#000000" />
-    <meta
-      name="description"
-      content="Web site created using create-react-app"
-    />
+    <meta name="description" content="Web site created using create-react-app" />
```
Comment (BowTiedDeployer) [inline public/index.html]: is this content field updated later? it should be `content="Stacks Decentralized Pools"`

File: src/components/LeftPanel.tsx [hunk 2]
```diff
@@ -65,12 +70,13 @@ const LeftPanel = ({ currentTheme }: ConnectWalletProps) => {
     <Box
       sx={{ width: anchor === 'top' || anchor === 'bottom' ? 'auto' : 250, height: '100%' }}
       role="presentation"
-      style={{ backgroundColor: colors[currentTheme].accent2 }}
+      style={{ backgroundColor: colors[appCurrentTheme].accent2, color: colors[appCurrentTheme].colorWriting }}
+      // style={{ backgroundColor: colors[currentTheme].accent2, color: colors[currentTheme].colorWriting }}
```
Comment (BowTiedDeployer) [inline src/components/LeftPanel.tsx]: TODO: remove this line

File: src/components/TableCreation.tsx [hunk 3]
```diff
@@ -13,8 +13,11 @@ import { TableSortLabel } from '@mui/material';
 import useCurrentTheme from '../consts/theme';
 import colors from '../consts/colorPallete';
 import { AllTableData } from '../consts/tableData';
+import '../components/appMenuSections/miningPool/styles.css';
+import { useAppSelector } from '../redux/store';
+import { selectCurrentTheme } from '../redux/reducers/user-state';
 
-const VirtuosoTableComponents: TableComponents<any> = {
+const VirtuosoTableComponents: TableComponents<any /*don't know the type here*/> = {
```
Comment (BowTiedDeployer) [inline src/components/TableCreation.tsx]: do we know it on a later commit?

File: src/components/appMenuSections/dashboard/styles.css
Comment (BowTiedDeployer) [inline src/components/appMenuSections/dashboard/styles.css:1]: this has no utility here

File: src/components/reusableComponents/profile/MoreInfoAboutContainerMiner.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  currentBalance: number;
+  totalWithdrawals: number | null;
+}
+
+const MoreInfoAboutContainerMiner = ({ currentBalance, totalWithdrawals }: IMinerMoreInfoAboutContainer) => {
+  const [exchange, setExchange] = useState<boolean | null>(false);
+  const { currentTheme } = useCurrentTheme();
+
+  const userSession = useAppSelector(selectUserSessionState);
+  const userAddress = userSession.loadUserData().profile.stxAddress.testnet;
```
Comment (BowTiedDeployer) [inline src/components/reusableComponents/profile/MoreInfoAboutContainerMiner.tsx]: should be modular network

mainnet -> .mainnet
testnet/devnet -> .testnet

File: src/components/reusableComponents/profile/MoreInfoAboutContainerPendingMiner.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+import { useAppSelector } from '../../../redux/store';
+import { readOnlyGetRemainingBlocksJoin } from '../../../consts/readOnly';
+import { ContractAddPending } from '../../../consts/smartContractFunctions';
+import colors from '../../../consts/colorPallete';
+import useCurrentTheme from '../../../consts/theme';
+
+const MoreInfoAboutContainerPendingMiner = () => {
+  const currentRole: UserRole = useAppSelector(selectCurrentUserRole);
+  const userSession = useAppSelector(selectUserSessionState);
+  const userAddress = userSession.loadUserData().profile.stxAddress.testnet;
```
Comment (BowTiedDeployer) [inline src/components/reusableComponents/profile/MoreInfoAboutContainerPendingMiner.tsx]: same here 
> should be modular network
> 
> mainnet -> .mainnet
> testnet/devnet -> .testnet

File: src/redux/reducers/index.ts [hunk 6]
```diff
@@ -1,19 +1,23 @@
...
+
 const initialState: IinitialState = {
   userState: defaultUserState,
+  theme: 'light',
 };
 
 interface IreduxAction {
   type: string;
-  payload?: any;
+  payload?: any; // know the type?
```
Comment (BowTiedDeployer) [inline src/redux/reducers/index.ts]: TODO: change type

File: src/redux/reducers/user-state.ts [hunk 7]
```diff
@@ -4,6 +4,7 @@ import { IinitialState } from '.';
 export interface IUserState {
   userSession: UserSession;
   userRole: UserRole;
+  // theme: Theme;
```
Comment (BowTiedDeployer) [inline src/redux/reducers/user-state.ts:7]: shouldn't line 2 `import { IinitialState } from '.';` be `from './index';`?

**Review by Alexandrescu — APPROVED** [review]

**Pull request merged**